ts/Series/PointAndFigure/PointAndFigureSeries.ts

```diff
@@ -0,0 +1,361 @@
+/* *
+ *
+ *  (c) 2010-2024 Kamil Musialowski
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *  Imports
+ *
+ * */
+
+import PointAndFigurePoint from './PointAndFigurePoint.js';
+import PointAndFigureSeriesDefaults from './PointAndFigureSeriesDefaults.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+
+import type Point from '../../Core/Series/Point.js';
+import type SVGAttributes from '../../Core/Renderer/SVG/SVGAttributes.js';
+import type Axis from '../../Core/Axis/Axis.js';
+import type PointAndFigureSeriesOptions from './PointAndFigureSeriesOptions';
+import type PointAndFigurePointOptions from './PointAndFigurePointOptions.js';
+
+import H from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+const { composed } = H;
+const {
+    scatter: ScatterSeries,
+    column: {
+        prototype: columnProto
+    }
+} = SeriesRegistry.seriesTypes;
+const {
+    addEvent,
+    extend,
+    merge,
+    pushUnique,
+    isNumber,
+    relativeLength
+} = U;
+
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+
+/* *
+ *
+ *  Functions
+ *
+ * */
+
+/**
+ *
+ */
+function generatePnfData(
+    this: Axis
+): void {
+    const series = this.series;
+
+    series.forEach((series): void => {
+        if (series.is('pointandfigure')) {
+            const pnfSeries = series as PointAndFigureSeries;
+            pnfSeries.generatePnfData();
+        }
+    });
+}
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The series type
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.pointandfigure
+ *
+ * @augments Highcharts.Series
+ */
+class PointAndFigureSeries extends ScatterSeries {
+
+    /* *
+     *
+     *  Static Properties
+     *
+    * */
+
+    public static defaultOptions: PointAndFigureSeriesOptions = merge(
+        ScatterSeries.defaultOptions,
+        PointAndFigureSeriesDefaults
+    );
+
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    public static compose(
+        AxisClass: typeof Axis
+    ): void {
+        if (pushUnique(composed, 'pointandfigure')) {
+            addEvent(AxisClass, 'postProcessData', generatePnfData);
```

Comment:
this event is fired on Axis, and just before that, there is a function `series.processData(hasExtremesChanged)`fired.
 Instead of creating such composition, can't we just extend `series.processData` ? The only thing that might be relevant, is that `applyGrouping` fires before, but this series is not using dataGrouping anyway.
 
summon @pawelfus

Reply:
Nice catch! If no errors, then refactor 👍